### ISSUE TITLE
Backport: UI for Allow S3 Snapshot service to trust internally signed certs when talking to the S3 endpoint

### ISSUE THREAD

**Issue by alena1108** [rancher/rancher#21104]
This ticket is for UI component of https://github.com/rancher/rancher/issues/21103 > we need to add one extra optional field in the snapshot configuration UI

@westlywright: @moelsayed worked on the backend part in case you have any questions.

**Comment by westlywright** [comment]
Available in UI Tag `2.2.81`

**Comment by sowmyav27** [comment]
verified on `rancher:v2.2-head` with UI tag - `2.2.81`
UI has a new field - `Custom CA Certificate`

**Scenario 1:**
Valid and correct certificate and S3 enabled with minio
Able to take snapshots and restore from snapshots

**Scenario 2:**
InValid certificates and S3 enabled with minio
Error on the UI, when the user clicks on create cluster or Save on an edit cluster.
```
Failed to validate cluster: invalid S3 endpoint CA certificate: data does not contain any valid RSA or ECDSA certificates
```

**Scenario 3:**
Valid and incorrect certificate and S3 enabled with minio
Error in the rancher logs on taking a snapshot on the UI
```
2019/06/27 20:52:16 [ERROR] ClusterController c-jgv8g [cluster-provisioner-controller] failed with : Failed to download etcd snapshot from s3, exit code [1]: time="2019-06-27T20:52:14Z" level=fatal msg="failed to set s3 server: failed to check s3 bucket:sowmya-test, err:Get https://x.x.x.x:9000/sowmya-test/?location=: x509: certificate signed by unknown authority" 
```
And the snapshot is seen stuck in `Activating` state
